lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -489,16 +489,28 @@ def commit(action, post, options, authorization = nil, method = :post)
 
         succeeded = success_from(action, response)
 
-        response(action, succeeded, response, options, source_id)
+        response(action, succeeded, response, options)
       end
 
-      def response(action, succeeded, response, options = {}, source_id = nil)
+      def commit(action, post, options, authorization = nil, method = :post)
+        if @options[:secret_key] || access_token_valid?
+          peform_payin(action, post, options, authorization, method)
+        else
+          MultiResponse.run do |r|
+            r.process { setup_access_token }
+            r.process { peform_payin(action, post, options, authorization, method) }
+          end
+        end
+      end
+
+      def response(action, succeeded, response, options = {})
         authorization = authorization_from(response) unless action == :unstore
         body = action == :unstore ? { response_code: response.to_s } : response
+
         Response.new(
           succeeded,
           message_from(succeeded, response, options),
-          body,
+          body.merge(access_token: @options[:access_token]),
```

Comment:
I think you'll also want to add expires to this so that we can access it in Core

Reply:
Yes, I will :)

Reply:
Actually since you're doing a MultiResponse you don't need to this merge since you'll be able to access the access_token and expires_in in Core

Reply:
good catch, done!